### ISSUE TITLE
jOOQ can't generate tables classes when two table valued functions have the same name

### ISSUE THREAD

**Issue by ssssteffff** [jOOQ/jOOQ#6896]
### Expected behavior and actual behavior:
jOOQ can't generate classes when 2 functions have the same name (yet different signature). Formerly (3.8.2), it used to generate several classes for routines (routines/Test1.java and routines/Test2.java) without records.
From jOOQ 3.9.0, it generates a unique "table" class (tables/Test.java) and record (tables/records/TestRecord.java) with duplicated fields in both class.

I get same behavior with versions 3.9.0, 3.10.0, 3.10.1 and 3.10.2.

### Steps to reproduce the problem:
Create two functions:

```
CREATE FUNCTION test()
  RETURNS TABLE(id BIGINT) AS $$
SELECT 1
$$ LANGUAGE SQL;

CREATE FUNCTION test(t INTEGER)
  RETURNS TABLE(id INTEGER) AS $$
SELECT t
$$ LANGUAGE SQL;
```

And apply codegen, you'll get:

- 3.8.2:
  - routines/Test1.java
  - routines/Test2.java
- 3.9.0 to 3.10.2
  - tables/records/TestRecord.java
  - tables/Test.java

### Versions:

- jOOQ: 3.9.x, 3.10.x
- Java: 8
- Database (include vendor): PostgreSQL 9.4
- JDBC Driver (include name if inofficial driver): PostgreSQL 42.1.4

**Comment by ssssteffff** [comment]
Ok, I read about [table valued functions](https://www.jooq.org/doc/3.10/manual/code-generation/codegen-advanced/codegen-config-table-valued-functions/). It seems that default behavior has changed, and it is now enabled (at least for Postgres databases). Disabling it, I get previous behavior back, so I can migrate to 3.10.2.

However, I did not know about this flag, and "blamed" jOOQ for this because I did scrap code to join on this table valued functions. Since I discovered jOOQ actually handles them, I would be happy to use them as Tables. If only jOOQ codegen would handle two table-valued functions using the same name ;-)
